lib/urllib.js

```diff
@@ -286,7 +286,9 @@ function requestWithCallback(url, args, callback) {
   };
   if (args.headers) {
     for (var k in args.headers) {
-      options.headers[k] = args.headers[k];
+      if (args.headers[k] !== null) {
```

Comment:
How about `undefined`? Should we ignore it too?

Reply:
@fengmk2 `undefined` is the default, so that might lead to side effects. A lookup for a header that doesn't exist returns `undefined`. Setting it to `null` is a more correct solution.